### ISSUE TITLE
Casting from timestamp[us] to timestamp[ns] out of bounds

### ISSUE THREAD

**Issue by imanebosch** [awslabs/aws-data-wrangler#427]
After executing 
```
wr.db.read_sql_query(
    "SELECT * FROM XXX_XXX", 
    con=eng_postgresql
)
```
I get the following error:

`ArrowInvalid: Casting from timestamp[us] to timestamp[ns] would result in out of bounds timestamp: 253402214400000000`

It happens with when `valid_to` column  is equal to `9999-12-31 00:00:00:0000`. Is there any method to avoid this error?

**Comment by igorborgest** [comment]
Thanks for reporting it.
Could you provide the whole exception traceback error message (**_cleaning up any sensitive information_**), please?

By now PyArrow and subsequently Wrangler does not accept this range of timestamp (_year=9999_) but we can add a flag to force the read even if it means to corrupt this value ([Like in this other case](https://github.com/awslabs/aws-data-wrangler/issues/296)). What do you think?

**Comment by imanebosch** [comment]
Hi,

Find below the whole traceback errror message.

For our use case, values `valid_to= 9999-12-31 00:00:00:0000` are used within the database for temporal modeling. Along with `valid_from`, which will always have non-problematic values, they are used to indicate that the record is valid from a particular day till today. So, it **would be fine to force the read since we do not mind the value itself, but its meaning**.

We have realized, thought, that if we use pandas directly, all seems to work fine.  

`pd.read_sql_query(f"SELECT * FROM {table}", con=eng_postgresql)`

Is there any drawback when using this approach?

`---------------------------------------------------------------------------
ArrowInvalid                              Traceback (most recent call last)
<ipython-input-4-efb25fc469df> in <module>
      1 df = wr.db.read_sql_query(
      2     "SELECT * FROM pay_ins_invoice_status",
----> 3     con=eng_postgresql
      4 )

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/awswrangler/db.py in read_sql_query(sql, con, index_col, params, chunksize, dtype)
    417         cursor = _con.execute(*args)
    418         if chunksize is None:
--> 419             return _records2df(records=cursor.fetchall(), cols_names=cursor.keys(), index=index_col, dtype=dtype)
    420         return _iterate_cursor(
    421             cursor=cursor, chunksize=chunksize, cols_names=cursor.keys(), index=index_col, dtype=dtype

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/awswrangler/db.py in _records2df(records, cols_names, index, dtype)
    198         integer_object_nulls=False,
    199         date_as_object=True,
--> 200         types_mapper=_data_types.pyarrow2pandas_extension,
    201     )
    202     if index is not None:

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/pyarrow/array.pxi in pyarrow.lib._PandasConvertible.to_pandas()

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/pyarrow/table.pxi in pyarrow.lib.Table._to_pandas()

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/pyarrow/pandas_compat.py in table_to_blockmanager(options, table, categories, ignore_metadata, types_mapper)
    777     _check_data_column_metadata_consistency(all_columns)
    778     columns = _deserialize_column_index(table, all_columns, column_indexes)
--> 779     blocks = _table_to_blocks(options, table, categories, ext_columns_dtypes)
    780 
    781     axes = [columns, index]

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/pyarrow/pandas_compat.py in _table_to_blocks(options, block_table, categories, extension_columns)
   1113     columns = block_table.column_names
   1114     result = pa.lib.table_to_blocks(options, block_table, categories,
-> 1115                                     list(extension_columns.keys()))
   1116     return [_reconstruct_block(item, columns, extension_columns)
   1117             for item in result]

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/pyarrow/table.pxi in pyarrow.lib.table_to_blocks()

~/Documents/repositories/cleverea-aws-batch/venv-dev/lib/python3.7/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowInvalid: Casting from timestamp[us] to timestamp[ns] would result in out of bounds timestamp: 253402214400000000`

**Comment by igorborgest** [comment]
Hi @imanebosch !

We added the `safe` argument to `wr.db.read_sql_query()`/`wr.db.read_sql_table()` to force the input of values out of bound. This values will be corrupted and became different values. But the rest will stay the same.

Do you mind to install it directly from our repository and give some feedback?
The idea is to publish it in the version `1.10.0` this weekend.

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git --no-use-pep517
```

**Comment by igorborgest** [comment]
Released on [1.10.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.10.0)
